cg/store/crud/read.py

```diff
@@ -1751,6 +1751,15 @@ def get_orders_by_workflow(
         )
         return orders.limit(limit).all()
 
+    def get_order_by_id(self, order_id: int) -> Order:
+        """Returns the entry in Order matching the given id."""
+        orders: Query = self._get_query(table=Order)
+        order_filter_functions: list[Callable] = [OrderFilter.FILTER_ORDERS_BY_ID]
+        orders: Query = apply_order_filters(
+            orders=orders, filter_functions=order_filter_functions, id=order_id
+        )
+        return orders.first()
```

Comment:
Side note: do we want to raise an exception in the database layer when an entity like this is not found? I think providing an id for an entry and not finding it warrants raising an `OrderNotFound` exception. That exception should be caught in the api layer and be used to return a 404